enterprise/models.py

```diff
@@ -713,6 +713,37 @@ def notify_enrolled_learners(self, catalog_api_user, course_id, users, admin_enr
         )
         send_enterprise_email_notification.delay(self.uuid, admin_enrollment, email_items)
 
+    def toggle_universal_link(self, enable_universal_link, link_expiration_date):
+        """
+        Sets enable_universal_link
+
+        If there is no change to be made, return
+
+        When enable_universal_link changes to;
+            True: a new link is created
+            False: all links are deactivate
+        """
+        if self.enable_universal_link == enable_universal_link:
+            return
+
+        self.enable_universal_link = enable_universal_link
+        self.save()
+
+        # If universal link is being disabled
+        if not enable_universal_link:
+            # Deactivate all EnterpriseCustomerInviteKey
+            EnterpriseCustomerInviteKey.objects.filter(
+                enterprise_customer=self,
+                is_active=True,
+            ).update(is_active=False)
+        # If universal link is being enabled and a date is passed
+        elif bool(link_expiration_date):
```

Comment:
I think you need to check if any `EnterpriseCustomerInviteKeys` already exist when it's being toggled to `True` - if they do, we wouldn't want to create a new one, right?

Reply:
Ah, that's right - the links active state is immutable!  So yeah, leave what you have. 👍